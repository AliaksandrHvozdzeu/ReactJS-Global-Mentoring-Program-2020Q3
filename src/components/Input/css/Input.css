.visibility {
    visibility: hidden;
    height: 0;
}
