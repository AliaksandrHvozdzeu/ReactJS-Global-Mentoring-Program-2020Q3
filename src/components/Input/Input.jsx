import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

export default function Input({ id, title, visibility, value, onChange }) {

  return (
    <>
      <label htmlFor={id} className={visibility}>
        {title}
        <input type="text" id={id} className="input" value={value} onChange={onChange}/>
      </label>
    </>
  );

}

Input.propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  title: PropTypes.string.isRequired,
  visibility: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  onChange: PropTypes.func.isRequired,
};
