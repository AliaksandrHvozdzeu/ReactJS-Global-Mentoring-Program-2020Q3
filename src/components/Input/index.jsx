import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const Input = ({ id, title, visibility, value, onChange }) => (
  <label htmlFor={id} className={visibility}>
    {title}
    <input type="text" id={id} className="input" value={value} onChange={onChange}/>
  </label>
);

Input.propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  title: PropTypes.string.isRequired,
  visibility: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onChange: PropTypes.func.isRequired
};

export default Input;
