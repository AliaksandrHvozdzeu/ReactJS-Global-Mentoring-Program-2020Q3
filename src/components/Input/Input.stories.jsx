import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import Input from './js/Input';
import '../../App.css';

export default {
  title: 'Input',
  decorators: [withKnobs],
};

export const Default = () => (
  <Input />
);
