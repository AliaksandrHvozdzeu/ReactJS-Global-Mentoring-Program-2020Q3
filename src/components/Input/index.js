import Input from './js/Input';

export default Input;
