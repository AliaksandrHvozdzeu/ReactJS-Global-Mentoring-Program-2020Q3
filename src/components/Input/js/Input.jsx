import React from 'react';
import PropTypes from 'prop-types';
import '../css/Input.css';

export default function Input({ id, title, visibility, value, onChange, name, type }) {

  return (
    <>
      <label htmlFor={id} className={visibility}>
        {title}
        <input id={id}
               type={type}
               className="input"
               value={value}
               name={name}
               readOnly={false}
               onChange={onChange}
               required/>
      </label>
    </>
  );

}

Input.propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  title: PropTypes.string.isRequired,
  visibility: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string,
  type: PropTypes.string,
};

Input.defaultProps = {
  value: '',
  name: '',
}
