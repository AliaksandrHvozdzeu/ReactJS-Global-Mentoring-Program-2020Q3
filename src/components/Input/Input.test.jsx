import React from 'react';
import rerender from 'react-test-renderer';
import Input from './js/Input';

describe('when Input', () => {
  test('then snapshot created', () => {
    const id = 'id';
    const title = 'title';
    const visibility = 'visibility';
    const value = 'value';
    const onChange = () => {
    };
    const name = 'name';
    const type = 'type';
    const onBlur = () => {
    };
    const error = 'error';
    const component = rerender.create(<Input id={id}
                                             title={title}
                                             visibility={visibility}
                                             value={value}
                                             onChange={onChange}
                                             name={name}
                                             type={type}
                                             onBlur={onBlur}
                                             error={error}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
