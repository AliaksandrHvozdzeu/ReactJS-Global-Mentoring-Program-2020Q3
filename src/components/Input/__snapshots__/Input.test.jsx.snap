// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Input then snapshot created 1`] = `
Array [
  <label
    className="visibility"
    htmlFor="id"
  >
    title
    <input
      className="input"
      id="id"
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      readOnly={false}
      type="type"
      value="value"
    />
  </label>,
  <p
    className="input-error"
  >
    error
  </p>,
]
`;
